### PR TITLE
Improving sparse tests.

### PR DESCRIPTION
**Pull request by drivanov**
## Description

Suppression of the following warnings that appear in tests implemented in `sparse/test_matmul.py` 
```
tests/python/pytorch/sparse/test_matmul.py::test_spspmm[1-1-3-shape_n_m0-rand_coo-rand_coo]
  /opt/dgl/dgl-source/tests/python/pytorch/sparse/test_matmul.py:101: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
    torch_A3 = torch.sparse.mm(torch_A1, torch_A2)
```
and `sparse/test_sparse_matrix.py`
```
tests/python/pytorch/sparse/test_sparse_matrix.py::test_torch_sparse_csr_conversion[shape0-indices0-indptr0]
  /opt/dgl/dgl-source/tests/python/pytorch/sparse/test_sparse_matrix.py:583: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
    torch_sparse_csr = torch.sparse_csr_tensor(
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by frozenbugs** [comment]
@dgl-bot

File: tests/python/pytorch/sparse/test_matmul.py [hunk 1]
```diff
@@ -98,7 +98,9 @@ def test_spspmm(create_func1, create_func2, shape_n_m, shape_k, nnz1, nnz2):
 
     torch_A1 = sparse_matrix_to_torch_sparse(A1)
     torch_A2 = sparse_matrix_to_torch_sparse(A2)
-    torch_A3 = torch.sparse.mm(torch_A1, torch_A2)
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=UserWarning)
+        torch_A3 = torch.sparse.mm(torch_A1, torch_A2)
```
Comment (frozenbugs) [inline tests/python/pytorch/sparse/test_matmul.py:103]: This is weird, torch_A1 and A2 are both sparse_coo_tensor, why the "UserWarning: Sparse **CSR tensor** support is in beta state." is triggered?

https://github.com/dmlc/dgl/blob/976970554d95d96b802faa32488d4bf836678340/tests/python/pytorch/sparse/utils.py#L153C17-L153C34

Comment (frozenbugs) [reply tests/python/pytorch/sparse/test_matmul.py:103]: I don't see this warning when I run this locally.

Comment (frozenbugs) [reply tests/python/pytorch/sparse/test_matmul.py:103]: I don't see this in my local run and also CI, are you referring to:

tests/python/pytorch/sparse/test_sparse_matrix.py::test_torch_sparse_csr_conversion[shape0-indices0-indptr0]
  /root/jenkins/workspace/dgl_PR-6147@2/tests/python/pytorch/sparse/test_sparse_matrix.py:583: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at ../aten/src/ATen/SparseCsrTensorImpl.cpp:54.)
    torch_sparse_csr = torch.sparse_csr_tensor(

Comment (drivanov) [reply tests/python/pytorch/sparse/test_matmul.py:103]: Actually, I see similar warnings in both places:
```
tests/python/pytorch/sparse/test_matmul.py::test_spspmm[1-1-3-shape_n_m0-rand_coo-rand_coo]
  /opt/dgl/dgl-source/tests/python/pytorch/sparse/test_matmul.py:101: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
    torch_A3 = torch.sparse.mm(torch_A1, torch_A2)
```
and 
```
tests/python/pytorch/sparse/test_sparse_matrix.py::test_torch_sparse_csr_conversion[shape0-indices0-indptr0]
  /opt/dgl/dgl-source/tests/python/pytorch/sparse/test_sparse_matrix.py:583: UserWarning: Sparse CSR tensor support is in beta state. If you miss a functionality in the sparse tensor support, please submit a feature request to https://github.com/pytorch/pytorch/issues. (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/SparseCsrTensorImpl.cpp:53.)
    torch_sparse_csr = torch.sparse_csr_tensor(
```
I just missed this warning `sparse/test_sparse_matrix.py` and I will fix it in a separate PR.

**Review by frozenbugs — APPROVED** [review]

**Pull request merged**